training-tracker-service/src/main/java/org/tolking/repository/TrainerSummaryRepository.java

```diff
@@ -0,0 +1,14 @@
+package org.tolking.repository;
+
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
+import org.tolking.entity.TrainerSummary;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
```

Comment:
there's no need to anotate inheritants for `Repository` classes. They will always be beans. `@Repository` is meant to be used without Spring Data, for exaple when you just use simple jdbc or Spring's JDBCTemplate. That annotation will create bean AND add several default exception handlers (methods will be wrapped with try-catch by spring)